src/findone-parallel-performance.c

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright 2021-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* Tests performance of parallel find operations on a client pool.
+ * The task definition mirrors the workload identified in CDRIVER-4002.
+ * The task definition is not part of the "MongoDB Driver Performance
+ * Benchmarking" specification. */
+
+#include "mongo-c-performance.h"
+
+#include <bson.h>
+#include <mongoc.h>
+
+#include <pthread.h>
+
+typedef struct {
+   pthread_t thread;
+   mongoc_client_t *client;
+   int n_findone_to_run;
+} findone_parallel_thread_context_t;
+
+typedef struct {
+   perf_test_t base;
+   mongoc_client_pool_t *pool;
+   int n_threads;
+   findone_parallel_thread_context_t *contexts;
+} findone_parallel_perf_test_t;
+
+/* FINDONE_FILTER_SIZE is the size of the BSON document {"id": 0}.
+ * Determined with this Python script:
+ *    import bson
+ *    print (len(bson.encode({"id": 0}))) # prints "13"
+ */
+#define FINDONE_FILTER_SIZE 13
```

Comment:
nit: Even though they're more customary in C, consider a const int instead of a macro, as scoping and other language rules will be obeyed.

Reply:
Done.